readthedocs/search/views.py

```diff
@@ -64,7 +66,15 @@ def elastic_search(request, project_slug=None):
         }
     )
 
-    results = None
+    serializers = collections.defaultdict(
+        lambda: ProjectSearchSerializer,
+        {
+            'project': ProjectSearchSerializer,
+            'file': PageSearchSerializer,
+        }
+    )
```

Comment:
I have no idea what this is doing, there must be clearer logic that expresses the same thing.